narwhal/consensus/src/bullshark.rs

```diff
@@ -32,6 +29,9 @@ pub mod randomized_tests;
 /// and consequently a commit.
 #[derive(Default)]
 pub struct LastRound {
+    /// The elected leader of the round. That's irrespective of whether the corresponding certificate
+    /// is found or not.
+    leader: Option<Authority>,
```

Comment:
Why is this an option? Won't we always elect a leader each round.

Reply:
That's mostly because when the protocol starts we don't have a "previous" leader , that's why nothing exists as default. Alternatively we could make the `LastRound` an `Option` as a whole, but I did consider making just the `leader` here instead.